web/src/features/panels/asset/SolarAssetPanel.tsx

```diff
@@ -0,0 +1,207 @@
+  const selectedAsset = useAtomValue(selectedSolarAssetAtom);
+  const setSelectedAsset = useSetAtom(selectedSolarAssetAtom);
+  const navigate = useNavigateWithParameters();
+  const location = useLocation();
+
+  if (!selectedAsset) {
+    return null;
+  }
+
+  console.log('random');
+  const { properties } = selectedAsset;
+
+  // Debug all properties to find potential URL fields
+  console.log('[SolarAssetPanel] All properties:', properties);
+  const possibleUrlFields = Object.entries(properties).filter(
+    ([_, value]) =>
+      typeof value === 'string' &&
+      (value.startsWith('http://') ||
+        value.startsWith('https://') ||
+        value.startsWith('www.'))
+  );
+  console.log('[SolarAssetPanel] Possible URL fields:', possibleUrlFields);
+
+  const handleClose = () => {
+    setSelectedAsset(null);
+    // Check if we are on a path that implies an asset is selected in the URL structure.
+    // This logic might need adjustment depending on final routing for assets.
+    const isAssetPath =
+      location.pathname.includes('/solar-asset/') ||
+      location.pathname.includes('/zone/solar-asset-');
+
+    if (isAssetPath) {
+      // Navigate to the base map view, preserving current map parameters if possible
+      // This part is a simplified navigation, assuming we want to go back to /map
+      // and that navigate hook handles existing params like timeRange, resolution.
+      navigate({ to: '/map', keepHashParameters: true });
+    }
+  };
+
+  const name = properties.name || properties.ASSET_NAME || 'Unnamed Asset';
+  const capacityMw = Number.parseFloat(String(properties.capacity_mw));
+  const source = properties.source || 'N/A';
+
+  // Try to find a URL from various possible fields
+  let sourceUrl = properties.source_url || properties.sourceUrl || properties.url || null;
+
+  // If source itself is a URL, use it
+  if (
+    typeof source === 'string' &&
+    (source.startsWith('http://') || source.startsWith('https://'))
+  ) {
+    sourceUrl = source;
+  }
+
+  // Handle URLs that start with www. but don't have a protocol
+  if (typeof sourceUrl === 'string' && sourceUrl.startsWith('www.')) {
+    sourceUrl = 'https://' + sourceUrl;
+  }
+
+  // For debugging - explicitly check if the URL is valid
+  let isValidUrl = false;
+  try {
+    if (sourceUrl) {
+      new URL(sourceUrl); // This will throw if the URL is invalid
+      isValidUrl = true;
+    }
+  } catch (error) {
+    console.error('[SolarAssetPanel] Invalid URL:', sourceUrl, error);
+  }
+
+  console.log('[SolarAssetPanel] Final sourceUrl:', sourceUrl, 'isValid:', isValidUrl);
+
+  // Hardcoded fallback URL for now - remove this when real URLs are working
+  if (!isValidUrl) {
+    sourceUrl =
+      'https://www.globalenergymonitor.org/projects/global-solar-power-tracker/';
+    isValidUrl = true;
+  }
+
+  const commissionYear = properties.commission_year
+    ? String(Math.floor(Number(properties.commission_year)))
```

Comment:
Same here about the conversion?

Why do we need to floor a year, it should already be an int and can either be turned into a string directly?